mathphys/imgproc.py

```diff
@@ -0,0 +1,1173 @@
+            data[data > saturation_threshold] = saturation_threshold
+        return data, indcsx, indcsy
+
+    @staticmethod
+    def normalize(data, maxval):
+        """."""
+        # benchmark for size=(1024, 1280)
+        #   2.91 ms ± 97.8 µs per loop
+        #   (mean ± std. dev. of 7 runs, 100 loops each)
+        # benchmark for size=1280
+        #   9.28 µs ± 407 ns per loop
+        #   (mean ± std. dev. of 7 runs, 100000 loops each)
+
+        new_data = data * (maxval/data.max())
+        return new_data
+
+    @staticmethod
+    def fit_gaussian(proj, indcs, center, offset):
+        """."""
+        indc = indcs - center  # centered fitting
+        proj = proj.copy() - offset
+        sel = proj > 0  # fit only positive data
+        vecy, vecx = proj[sel], indc[sel]
+        logy = _np.log(vecy)
+        pfit = _np.polyfit(vecx, logy, 2)
+        if pfit[0] < 0:
+            sigma = _np.sqrt(-1/pfit[0]/2) if pfit[0] < 0 else 0.0
+            mu = pfit[1] * sigma**2
+            amp = _np.exp(pfit[2] + (mu/sigma)**2/2)
+            mu += center
+        else:
+            sigma, mu, amp, offset = [float('nan')] * 4
+        return sigma, mu, amp, offset
+
+    @staticmethod
+    def calc_fit(image, proj, indcs, center):
+        """."""
+        # get roi gaussian fit
+        param = CurveFitGauss.fit_gaussian(
+            proj, indcs, center, image.intensity_min)
+        if param[0] > 0:
+            gfit, *_ = CurveFitGauss.generate_gaussian_1d(indcs, *param)
+            roi_gaussian_fit = gfit
+            error = _np.sum((gfit - proj)**2)
+            error /= _np.sum(proj**2)
+            roi_gaussian_error = 100 * _np.sqrt(error)
+        else:
+            roi_gaussian_fit = 0 * proj
+            roi_gaussian_error = float('nan')
+        fit = (param, roi_gaussian_fit, roi_gaussian_error)
+        return fit
+
+    @staticmethod
+    def _process_args(
+            indcs, sigma, mean, amplitude,
+            offset, rand_amplitude, saturation_threshold):
+        sigma = sigma or float('Inf')
+        if isinstance(indcs, (int, float)):
+            indcs = _np.arange(int(indcs))
+        elif isinstance(indcs, (tuple, list)):
+            indcs = _np.array(indcs)
+        elif isinstance(indcs, _np.ndarray):
+            indcs = _np.array(indcs)
+        else:
+            raise ValueError('Invalid indcs!')
+        if indcs.size < 2:
+            raise ValueError('Invalid indcs!')
+        elif indcs.size == 2:
+            indcs = _np.arange(*indcs)
+        amplitude = amplitude or 0
+        offset = offset or 0
+        rand_amplitude = rand_amplitude or 0
+        res = (
+            indcs, sigma, mean, amplitude,
+            offset, rand_amplitude, saturation_threshold
+            )
+        return res
+
+
+class ScipyFitGauss(CurveFitGauss):
+    """."""
+
+    def __init__(self, scipy_curv_fit_func):
+        """."""
+        self._scipy_curve_fitfunc = scipy_curv_fit_func
```

Comment:
i Think you could import scipy here, inside the `__init__` method.

Reply:
yes. I think it is cleaner this way...

Reply:
done